src/utils/options.ts

```diff
@@ -1,2 +1,20 @@
-export const defaultOptionValue = <O>(opt: O) => (opt as any).id;
-export const defaultOptionLabel = <O>(opt: O) => (opt as any).name;
+import { fail } from "src/utils";
+
+// This `any` is currently on purpose to ignore type errors in ChipSelectField
+export function defaultOptionValue<O>(opt: O): any {
+  if (typeof opt !== "object" || !opt) fail(`Option ${opt} has no id or code`);
+  // Use `in` because returning undefined is fine
+  return "id" in opt ? opt.id : "code" in opt ? opt.code : fail(`Option ${JSON.stringify(opt)} has no id or code`);
+}
+
+export function defaultOptionLabel<O>(opt: O): any {
+  if (typeof opt !== "object" || !opt) fail(`Option ${opt} has no id or code`);
```

Comment:
hmm... how destructive would it be to allow strings. Like, I'd _like_ to be able to do

```
options={["One", "Two", "Three"]}
// or
options={Object.values(SomeEnum)}
```

where ID/Val would be the same. Seems unnecessary to map an Enum over to an Object just for the sake of this. But, I'm sure a lot of infra expects an Object so makes sense it would be rough to support.

Reply:
Yeah, that's a fair ask. Will think about it...